kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/ProduceActionIntegrationTest.java

```diff
@@ -82,13 +87,54 @@ public class ProduceActionIntegrationTest {
   private static final String DEFAULT_VALUE_SUBJECT = "topic-1-value";
 
   @RegisterExtension
-  public final DefaultKafkaRestTestEnvironment testEnv = new DefaultKafkaRestTestEnvironment();
+  public final DefaultKafkaRestTestEnvironment testEnv = new DefaultKafkaRestTestEnvironment(false);
 
   @BeforeEach
-  public void setUp() throws Exception {
+  public void setUp(TestInfo testInfo) throws Exception {
+    Properties restConfigs = new Properties();
+    // Adding custom KafkaRestConfigs for individual test-cases/test-methods below.
+    if (testInfo.getDisplayName().contains("CallerIsRateLimited")) {
+      restConfigs.put(KafkaRestConfig.RATE_LIMIT_ENABLE_CONFIG, "true");
+      restConfigs.put(KafkaRestConfig.PRODUCE_RATE_LIMIT_ENABLED, "true");
+      restConfigs.put(KafkaRestConfig.RATE_LIMIT_BACKEND_CONFIG, "resilience4j");
+      // The happy-path testing, i.e. rest calls below threshold succeed are already covered by the
+      // other existing tests. The 4 tests below, 1 per rate-limit config, set a very low rate-limit
+      // of "1", to deterministically make sure limits apply and rest-calls see 429s.
+      if (testInfo
```

Comment:
I got caught out with .getDisplayName() previously, it's not actually the test method name, just make sure that this is the intended test info that you want to use

Reply:
Good call, to doubly sure, I am explicitly setting the @DisplayName().